Target-MVD-Mathias/AppDelegate.swift

```diff
@@ -79,18 +80,84 @@ class AppDelegate: UIResponder, UIApplicationDelegate, PushNotificationDelegate
     completionHandler(UIBackgroundFetchResult.noData)
   }
   
+  func onPushReceived(_ pushManager: PushNotificationManager!, withNotification pushNotification: [AnyHashable : Any]!, onStart: Bool) {
+    if let pushBody = pushNotification["u"] as? String {
+      let json = JSON(parseJSON: pushBody)
+      
+      if json["text"].string != nil {
+        processNewMessageNotification(json) { (match, message) in
+          self.newMessageNotificationReceived(match, message: message)
+        }
+      }
+    }
+  }
+  
   func onPushAccepted(_ pushManager: PushNotificationManager!, withNotification pushNotification: [AnyHashable : Any]!, onStart: Bool) {
     if let pushBody = pushNotification["u"] as? String {
       let json = JSON(parseJSON: pushBody)
       
       if json["title"].string != nil {
-        let match = Match.parse(fromJSON: json)
-        let chatViewController = UIStoryboard.instantiateViewController(ChatViewController.self)
-        window?.rootViewController?.present(chatViewController!, animated: false, completion: nil)
+        processNewMatchNotification(json)
+      } else if json["text"].string != nil {
```

Comment:
If you use `.string` and the result is nil, the app will crash. Try using `.stringValue` and if the result does not exist the value will be nil by default

Reply:
It's the other way around, `.string` is the optional value.